components/fxa-client/src/lib.rs

```diff
@@ -36,8 +37,9 @@ pub mod migrator;
 pub mod msg_types {
     include!("mozilla.appservices.fxaclient.protobuf.rs");
 }
-mod http_client;
-mod oauth;
+// TODO: figure out a way without making too much public....
+pub mod http_client;
```

Comment:
FWIW, I think it would be fine to make the `http_client` module public, and call its methods directly from the code for `TestAccount`. The risk of mis-use seems pretty low.

For methods that should really only be used from tests, I'm a fan of giving them super explicit names, so you can imagine the test code calling something like `client.test_only_account_create(email, password)` or similar.